tools/python/test_server/start_server.py

```diff
@@ -1,7 +1,10 @@
 #!/usr/bin/env python3
 
 import subprocess
+import time
 
 SERVER_RUNNABLE = 'server/testserver.py'
 
 subprocess.Popen(['python3', SERVER_RUNNABLE], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+
+time.sleep(2)  # Sleep for 2s to make sure the server is running
```

Comment:
please try to read from the pipe instead. sleep() is never a good option.

Reply:
Agree. No sleeps please. The code should wait only as much as necessary.